### PR TITLE
Fi 722 - removing tests for non uscdi

### PR DESCRIPTION
**Pull request by czh-orz**
This PR removes tests for delayed sequences. It now sets encounter as a delayed resource because it's not in USCDI.  Now delayed sequences will only have the read, profile validation, and reference resolution tests. It will do them on all references found to those resources from the prior searches.

**Submitter:**
- [X] This pull request describes why these changes were made
- [X] Internal ticket for this PR: https://oncprojectracking.healthit.gov/support/browse/FI-722
- [X] Internal ticket links to this PR
- [X] Internal ticket is properly labeled (Community/Program)
- [X] Internal ticket has a justification for its Community/Program label
- [X] Code diff has been reviewed for extraneous/missing code
- [X] Tests are included and test edge cases
- [X] Tests/code quality metrics have been run locally and pass

**Reviewer 1:**

Name:
- [x] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

**Reviewer 2:**

Name:
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure
      where appropriate, and accomplishes the task's purpose
- [ ] The tests appropriately test the new code, including edge cases
- [ ] You have tried to break the code

### CODE REVIEW

**Comment by arscan** [comment]
@czh-orz could you rebase and regenerate the tests.  Also the `onc_program_module.yml` file needs to be updated with the new order, I think?

**Comment by czh-orz** [comment]
rebased and fixed

**Comment by Jammjammjamm** [comment]
When I run the generator, I get different output than what is committed here. I think it is the same problem @okeefm encountered related to missing files required for generation.

File: lib/modules/uscore_v3.1.0/bodyheight_sequence.rb [hunk 1]
```diff
@@ -478,9 +478,78 @@ def patient_ids
         skip 'No Provenance resources were returned from this search' unless provenance_results.present?
       end
 
-      test :validate_resources do
+      test 'All must support elements are provided in the Observation resources returned.' do
```
Comment (Jammjammjamm) [inline lib/modules/uscore_v3.1.0/bodyheight_sequence.rb:481]: Is this test reordering intended?

**Comment by okeefm** [comment]
I re-added the missing files in my PR, so getting that merged would fix this.

File: lib/modules/uscore_v3.1.0/bodyheight_sequence.rb (hunk 1 above)
Comment (czh-orz) [reply lib/modules/uscore_v3.1.0/bodyheight_sequence.rb]: No, I changed it back.

**Comment by arscan** [comment]
> I re-added the missing files in my PR, so getting that merged would fix this.

Is this fixed @okeefm

**Comment by okeefm** [comment]
Not yet, #44 is still outstanding, but I'm rebasing and fixing the changes needed now that #45 has been merged. Then it should be ready to go.

**Comment by arscan** [comment]
I wonder if we need to better document why we aren't including PractitionerRole and Location (the answer: because there are no references to those US Core profiles within USCDI-mapped FHIR resources).  Because it *looks* like a gap.

I'm starting to think that what we want to do is have the US Core tests include them.  Then have an input that is simply uscdi_constrained, and if that is set by the tester it omits those tests.  And in the program edition we have the flag locked to true.

But I think we can start with this and discuss if that is the appropriate approach in the future.

**Pull request merged**